src/pages/Logs/Logs.tsx

```diff
@@ -1,9 +1,300 @@
+        : []
+      const number = toNumber(toggleValue[0] ?? 0)
+      const count = isEqual(toggleValue[1], 'hour') ? 'hour' : 'days'
+
+      const formattedCurrentDate = currentDate.format('DD/MM/YYYY')
+      const startDate = values?.date_range.start || formattedCurrentDate
+      const startTime = values?.time_range.start || currentTime
+      const endDate = values?.date_range.end || formattedCurrentDate
+      const endTime = values?.time_range.end || currentTime
+
+      if (!values?.last_date) {
+        setValue('date_range', { start: startDate, end: endDate })
+        setValue('time_range', { start: startTime, end: endTime })
+      }
+
+      const startDateTime = !!values?.last_date
+        ? currentDate.subtract(number, count)
+        : dayjs.utc(`${startDate} ${startTime}`, 'DD/MM/YYYY HH:mm:ss')
+
+      const endDateTime = !!values?.last_date
+        ? currentDate
+        : dayjs.utc(`${endDate} ${endTime}`, 'DD/MM/YYYY HH:mm:ss')
+
+      const payload = {
+        event_type: values?.activity,
+        text: values?.search,
+        department_id: values?.department_id,
+        start_datetime: startDateTime.format('YYYY-MM-DDTHH:mm:ss[Z]'),
+        end_datetime: endDateTime.format('YYYY-MM-DDTHH:mm:ss[Z]'),
+      }
+      setValue('payload', payload)
+
+      try {
+        await handleFilterChange(payload)
+      } catch (errorData) {
+        showValidationErrorMessage(errorData)
+      }
+    },
+    [currentDate, currentTime, handleFilterChange, setValue]
+  )
+
+  const { date_range, time_range, last_date, search } = watch()
+
+  useEffect(() => {
+    if (search && size(search) > 2) {
+      handleSubmit(onSubmit)()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [search])
+
+  useEffect(() => {
+    if (!isEmpty(date_range) || !isEmpty(time_range)) {
+      setValue('last_date', '')
+    }
+  }, [date_range, setValue, time_range])
+
+  useEffect(() => {
+    if (!!last_date) {
+      resetField('date_range')
+      resetField('time_range')
+    }
+  }, [last_date, resetField])
+
+  const handleExportLogs = useCallback(async () => {
+    const payload: AuditLogPayloadType = pickBy(
+      getValues('payload'),
+      (val) => !!val
+    )
+
+    try {
+      await exportCSV(payload)
+    } catch (errorData) {
+      showValidationErrorMessage(errorData)
+    }
+  }, [exportCSV, getValues])
+
+  return (
+    <>
+      <div className={classes.logsHeader}>
+        <h1>{t('menu.logs')}</h1>
+        <Tooltip helpSectionKey="logs" />
+      </div>
+
+      <Container className={classes.container}>
+        <div className={classes.buttonsContainer}>
+          <h4 className={classes.filters}>{t('logs.filters')}</h4>
+          <div className={classes.buttonsContainer}>
+            <Button
+              children={t('logs.clean_fields')}
+              ariaLabel={t('logs.clean_fields')}
+              size={SizeTypes.S}
+              onClick={() => reset()}
```

Comment:
or just onClick={reset}

Reply:
I usually use shorter version, but in that case this will give me the typescript error

Reply:
Okay true yeah, reset doesn't match the typing we have for onClick. Ideal solution would be to change the onClick typing a bit to allow for it, but let's keep it as is for now, it's not really important